examples/02-plot/element-picking.py

```diff
@@ -0,0 +1,69 @@
+"""
+.. _element_picking_example:
+
+Picking elements of a mesh
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+This example demonstrates how to pick different elements on meshes using
+:func:`enable_element_picking() <pyvista.Plotter.enable_element_picking>`.
+
+The different elements of a mesh are:
+
+* Mesh: pick the entire mesh (equivalent to :func:`enable_mesh_picking() <pyvista.Plotter.enable_mesh_picking>`.)
+* Cell: pick a cell of the mesh (equivalent to :func:`enable_cell_picking() <pyvista.Plotter.enable_cell_picking>`.)
+* Face: pick a single face of a cell on the mesh
+* Edge: pick a single edge of a cell on the mesh
+* Point: pick a single point on the mesh
```

Comment:
I also see a method `enable_rectangle_picking`, is there a reason it's left out of this group of examples?

Reply:
This is actually a bit different than `enable_point_picking`. `enable_point_picking` picks points under the cursor: sometimes in free space, sometimes on a mesh, and if the right picker is chosen, sometimes it will snap to points. `enable_element_picking` specifically snaps to points on a mesh. This is why I left it out.


`enable_rectangle_picking` isn't meant to be used directly, so I don't want to advertise it. `enable_rectangle_through_picking` and `enable_rectangle_visible_picking` are the interfaces meant to be used for this type of picking. Further, rectangle selection picks under some sort of area or volume whereas these methods are point-based picking